tools/generate-js.py

```diff
@@ -102,17 +102,66 @@ def generate_tlvtype(tlvtype: 'TlvMessageType'):
     # Now, generate table
     print('const tlv_{} = {{\n'.format(tlvtype.name))
     for f in tlvtype.fields:
-        print('    {num}: [ "{fname}", towire_{tlvname}_{fname}, fromwire_{tlvname}_{fname} ],\n'
+        print('    {num}: [ "{fname}", towire_{tlvname}_{fname}, fromwire_{tlvname}_{fname} ],'
               .format(num=f.number, tlvname=tlvtype.name, fname=f.name))
     print('}\n')
 
 
 def generate_msgtype(name: str):
-    raise RuntimeError("FIXME: Implement!")
-
+    print('function towire_{tlvname}(value)\n'
+              '{{'
+              .format(tlvname=name.name))
+    print('    let _n = 0;\n'
+          '    let buf = Buffer;')
+    for f in name.fields:
+        generate_towire_field(f, name.fields)
```

Comment:
messages are a bit different: they start with the number, as a 16 bit.  So you need to add a 'towire_u16()' at the start:

Perhaps `print('let buf = towire_u16({});'.format(name.number);`

Also, the variable `name: str` is really a `msgtype: pyln.proto.message.MessageType`.